lib/plugins/filter/template_locals/i18n.ts

```diff
@@ -13,7 +13,13 @@ function i18nLocalsFilter(locals) {
     const pattern = new Pattern(`${i18nDir}/*path`);
     const data = pattern.match(locals.path);
 
-    if (data && data.lang && i18nLanguages.includes(data.lang)) {
+    if (
+      typeof data !== 'undefined'
+      && !Array.isArray(data)
+      && typeof data !== 'boolean'
+      && data.lang
+      && i18nLanguages.includes(data.lang)
+    ) {
```

Comment:
This is somewhat over-complicated for a simple type guard. Is it possible to simplify it?

Reply:
> This is somewhat over-complicated for a simple type guard. Is it possible to simplify it?

because there has 3 return type (`boolean | RegExpMatchArray | Record<string, unknown>`)
see https://github.com/hexojs/hexo-util/pull/311/commits/e0719fb7386ce5c528caf6811aea53df259a48a8

see boolean return conflict
-    https://github.com/dimaslanjaka/hexo/blob/4ba63bc65d05d231fd1db1053013e2ae71661a70/lib/plugins/processor/common.ts#L29-L32
-    https://github.com/dimaslanjaka/hexo/blob/4ba63bc65d05d231fd1db1053013e2ae71661a70/lib/theme/processors/source.ts#L25-L32

ive tried:
- when only using negate conditional (`!data`), ts throw error `property lang does not exist on Array type`
- when only validate `!Array.isArray(data)`, ts throw error `property lang does not exist on boolean type`

Reply:
if there is any other suggestion, maybe it can be simplified without breaking (throw) during `npm run build`

Reply:
or. enable explicit any, down level iterator, and moved all export as ES Export style. might it can used square bracket (`data['page']`) to suppress error validation during `npm run build`

Reply:
sample supress syntax https://github.com/dimaslanjaka/static-blog-generator/blob/8cf940f1f7141085c42da874417c3d1a0cecf280/packages/sbg-utility/src/utils/scheduler.ts#L127

but these syntax style unsupported by hexo eslint config, it should be automatically fixed (moved to dots property style - `data.page`) on commit.